src/common/logger.js

```diff
@@ -66,48 +77,57 @@ export class Logger {
 		return errStr;
 	}
 
-	fmtMessage(msg) {
+	static checkLevels(level) {
+		for (let tlevel of [LOG_LEVEL_CONSOLE, LOG_LEVEL_FILE]) {
+			if (compareLevels(tlevel, level)) return true;
+		}
+	}
+
+	fmtMessage(level, msg, args) {
+		if (msg instanceof Error) {
+			msg = this.constructor.errToStr(msg);
+		}
+		args = args.map(e => (e instanceof Error ? this.constructor.errToStr(e) : e));
+		msg = vsprintf(msg, args);
 		if (!this.prefix) return msg;
-		return `${this.prefix} ${msg}`;
+		return `${level.toUpperCase()}: ${this.prefix} ${msg}`;
 	}
 
 	isFiltered(level, msg = '') {
 		return this.filter(level, msg);
 	}
 
-	logMessage(level, msg) {
-		if (msg instanceof Error) {
-			msg = this.constructor.errToStr(msg);
-		}
-
-		msg = this.fmtMessage(msg);
+	logMessage(level, msg, args) {
+		if (!this.constructor.checkLevels(level)) return;
```

Comment:
Is there some reason this wasn't incorporated in the existing filter function?

Reply:
yes, I am applying filter on a formatted string. I want to check levels before, to avoid formatting.